### DISCUSSION TITLE
Generate pydantic model code as a module

### DISCUSSION THREAD

**Discussion by TheCodingLand** [collerek/ormar#543] [Q&A]
Hello,

I was wondering if there is a way to inspect / generate the matching exact pydantic / BaseModel code from an ormar.Model.
The goal would be to generate a python library for use by third party apps, that do not need the ormar models themselves, but might need the code to query an API.

I'm building a platform where a lot of data exchange must be standardized, and I'd like to reuse the same pydantic models, so why not use my ormar models if it's possible ?

Example :

```python
class Item(ormar.Model):
    class Meta(BaseMeta):
        pass
    id = ormar.Integer(primary_key=True)
    name = ormar.String(max_length=200)
```

I'd like to generate the following code as as string to output in a python file:

```python
class Item(BaseModel):
    id : int
    name : str = Field(..., max_length=200)
``` 
Thanks !

**Comment by ponytailer** [comment]
```
schema = Item.get_pydantic()
pydantic_model = schema(id=1, name="123")

OR

class ItemExtra(item.get_pydantic()):
      label: str

```

**Reply by collerek** [reply]
Like @ponytailer wrote -> you can easily get a pedantic model with `get_pydantic()` but I don't think `pydantic` has some kind of Class `__repr__` that would print the code as a class definition so you would have to write your own.

I'm doing something very similar with ormar models created from sqlalchemy ones in https://github.com/collerek/sqlalchemy-to-ormar, so you can check it for inspiration. You simply need to write pydantic models class representations\definitions into a new python file.
